File: app/src/main/java/com/possible/tourrefactorsample/data/ControllerResult.java
package com.possible.tourrefactorsample.data;

import java.net.UnknownHostException;

public class ControllerResult<T> {
    private Throwable exception = null;
    private T result = null;

    public ControllerResult(T result) {
        this.result = result;
    }

    public ControllerResult(Throwable exception, T result) {
        this.exception = exception;
        this.result = result;
    }

    public Throwable getException() {
        return exception;
    }

    public T getResult() {
        return result;
    }

    public boolean isNetworkError() {
        return exception != null && exception instanceof UnknownHostException;
    }
}

File: app/src/main/java/com/possible/architecturesample/data/ControllerResult.kt
package com.possible.architecturesample.data

import java.net.UnknownHostException

class ControllerResult<T> {
    var exception: Throwable? = null
    lateinit var result: T

    constructor(result: T) {
        this.result = result
    }

    constructor(exception: Throwable, result: T) {
        this.exception = exception
        this.result = result
    }

    val isNetworkError: Boolean
        get() = exception != null && exception is UnknownHostException
}
